weather_report/main.py

```diff
@@ -0,0 +1,39 @@
+"""main.py
+"""
+import argparse
+
+from report_generator import ReportGenerator
+
+
+def main() -> None:
+    """Call respective functions based on inputs
+    """
+    parser = argparse.ArgumentParser(description="Weather Report file")
+    parser.add_argument(
+        "-e",
+        "--year",
+        type=int,
+        help="Takes year as input for year report"
+    )
+    parser.add_argument(
+        "-a",
+        "--month",
+        help="Year and month to generate month average report",
+    )
+    parser.add_argument(
+        "-c",
+        "--year_month",
+        help="Year and month to generate daily report of month with bars",
+    )
```

Comment:
Provide `metadata` argument also. It will help in defining the format of the input.